speechbrain/lobes/models/huggingface.py

```diff
@@ -0,0 +1,361 @@
+"""This lobe is the interface for huggingface transformers models
+It enables loading config and model via AutoConfig & AutoModel.
+
+Transformer from HuggingFace needs to be installed:
+https://huggingface.co/transformers/installation.html
+
+Authors
+ * Titouan Parcollet 2021, 2022, 2023
+ * Mirco Ravanelli 2021
+ * Boumadane Abdelmoumene 2021
+ * Ju-Chieh Chou 2021
+ * Artem Ploujnikov 2021, 2022
+ * Abdel Heba 2021
+ * Aku Rouhe 2022
+ * Arseniy Gorin 2022
+ * Ali Safaya 2022
+ * Benoit Wang 2022
+ * Adel Moumen 2022, 2023
+ * Andreas Nautsch 2022, 2023
+ * Luca Della Libera 2022
+ * Heitor Guimarães 2022
+ * Ha Nguyen 2023
+"""
+import os
+import torch
+import logging
+import pathlib
+from torch import nn
+from typing import Union
+from huggingface_hub import model_info
+from speechbrain.pretrained.fetching import fetch
+from speechbrain.dataio.dataio import length_to_mask
+
+
+# We check if transformers is installed.
+try:
+    from transformers import (
+        AutoConfig,
+        AutoTokenizer,
+        AutoFeatureExtractor,
+        AutoModelForPreTraining,
+        # AutoProcessor,
+        AutoModel,
+    )
+
+except ImportError:
+    MSG = "Please install transformers from HuggingFace to use wav2vec2 / Hubert\n"
+    MSG += "E.G. run: pip install transformers"
+    raise ImportError(MSG)
+
+logger = logging.getLogger(__name__)
+
+
+class HuggingFaceTransformer(nn.Module):
+    """This lobe provides AutoClass architecture loading into SpeechBrain modules.
+
+    See:
+    https://huggingface.co/docs/transformers/model_doc/auto
+    https://huggingface.co/docs/transformers/autoclass_tutorial
+
+    Arguments
+    ---------
+    source : str
+        HuggingFace hub name: e.g "facebook/wav2vec2-large-lv60"
+    save_path : str
+        save directory of the downloaded model.
+    tokenizer: bool (default: False)
+        If True, load the tokenizer (necessary for certain models, for example, mbart, whisper, etc.)
+    for_pretraining: bool (default: False)
+        If True, build the model for pretraining
+    freeze : bool (default: True)
+        If True, the model is frozen. If False, the model will be trained
+        alongside with the rest of the pipeline.
+    cache_dir: str or Path (default: None)
+        Location of HuggingFace cache for storing pre-trained models, to which symlinks are created.
+
+    Example
+    -------
+    >>> model_hub = "facebook/wav2vec2-base-960h"
+    >>> save_path = "tmp"
+    >>> model = HuggingFaceTransformer(model_hub, save_path=save_path)
+    """
+
+    def __init__(
+        self,
+        source,
+        save_path,
+        tokenizer=False,
+        for_pretraining=False,
+        freeze=False,
+        sampling_rate=16000,
```

Comment:
Do we need the sampling_rate here?

Reply:
I put it there because whisper puts it: https://github.com/speechbrain/speechbrain/blob/024478de11a75d8b8fa0dfbdd0cddcf5423e1358/speechbrain/lobes/models/huggingface_whisper.py#L100

Reply:
This could be solved by passing a **kwarg instead. This is to avoid having 1 billion arguments once we have many model implemented